src/ContactModels/src/ContinuousContactModel.cpp

```diff
@@ -0,0 +1,239 @@
+    m_isControlMatrixComputed = false;
+    m_isAutonomusDynamicsComputed = false;
+}
+
+void ContinuousContactModel::setImmutableParameters(
+    const std::unordered_map<std::string, std::any>& parameters)
+{
+    if (!getVariable(parameters, "length", m_length))
+        throw std::runtime_error("[ContinuousContactModel::setImmutableParameters] Unable to get "
+                                 "the variable named length");
+
+    if (!getVariable(parameters, "width", m_width))
+        throw std::runtime_error("[ContinuousContactModel::setImmutableParameters] Unable to get "
+                                 "the variable named width");
+
+    // the parameters has been update the previous quantities has to be evaluated again
+    m_isContactWrenchComputed = false;
+    m_isControlMatrixComputed = false;
+    m_isAutonomusDynamicsComputed = false;
+}
+
+ContinuousContactModel::ContinuousContactModel(
+    const std::unordered_map<std::string, std::any>& immutableParameters,
+    const std::unordered_map<std::string, std::any>& mutableParameters)
+    : m_twist(iDynTree::Twist::Zero())
+    , m_frameTransform(iDynTree::Transform::Identity())
+    , m_nullForceTransform(iDynTree::Transform::Identity())
+{
+    setMutableParameters(mutableParameters);
+    setImmutableParameters(immutableParameters);
+
+    // reset matrix and vectors
+    m_autonomousDynamics.zero();
+    m_controlMatrix.zero();
+}
+
+ContinuousContactModel::ContinuousContactModel(
+    const std::unordered_map<std::string, std::any>& parameters)
+    : m_twist(iDynTree::Twist::Zero())
+    , m_frameTransform(iDynTree::Transform::Identity())
+    , m_nullForceTransform(iDynTree::Transform::Identity())
+{
+    setMutableParameters(parameters);
+    setImmutableParameters(parameters);
+
+    // reset matrix and vectors
+    m_autonomousDynamics.zero();
+    m_controlMatrix.zero();
+}
+
+void ContinuousContactModel::computeContactWrench()
+{
+    double area = m_length * m_width;
+
+    auto force(iDynTree::toEigen(m_contactWrench.getLinearVec3()));
+    auto torque(iDynTree::toEigen(m_contactWrench.getAngularVec3()));
+
+    auto position(iDynTree::toEigen(m_frameTransform.getPosition()));
+    auto rotation(iDynTree::toEigen(m_frameTransform.getRotation()));
+
+    auto linearVelocity(iDynTree::toEigen(m_twist.getLinearVec3()));
+    auto angularVelocity(iDynTree::toEigen(m_twist.getAngularVec3()));
+
+    auto nullForcePosition(iDynTree::toEigen(m_nullForceTransform.getPosition()));
+    auto nullForceRotation(iDynTree::toEigen(m_nullForceTransform.getRotation()));
+
+    // compute the force
+    force = std::abs(rotation(2, 2)) * area * (m_springCoeff * (nullForcePosition - position)
+                                               - m_damperCoeff * linearVelocity);
+
+    // compute the torque
+    auto skewRe1 = iDynTree::skew(rotation.col(0));
+    auto skewRe2 = iDynTree::skew(rotation.col(1));
+    torque = std::abs(rotation(2, 2)) * area / 12 * (m_length * m_length
```

Comment:
Maybe it would be better to have a link to reference how this formula is obtained.

Reply:
I can update the `doxygen` documentation adding the explanation of the model